regression/cbmc/no_nondet_static/main.c

```diff
@@ -0,0 +1,94 @@
+int x;
+int y = 23;
+const int z = 23;
+
+struct s { int x; };
+struct s s1;
+struct s s2 = { 23 };
+const struct s s3 = { 23 };
+
+struct t { int x; const int y; };
+struct t t1;
+struct t t2 = { 23, 23 };
+const struct t t3 = { 23, 23 };
+
+struct u { const int x; };
+struct u u1;
+struct u u2 = { 23 };
+const struct u u3 = { 23 };
+
+struct list { const int datum; struct list * next; };
+
+// The point of this is that cbmc needs to correctly handle
+// the embedded datum int the nested struct which is const.
```

Comment:
This comment isn't quite true. The issue is not with the datum field, its with the recursive reference to the containing structure - i.e. `next`.